imap_processing/cli.py

```diff
@@ -241,67 +243,159 @@ def download_dependencies(self):
             file_list.append(imap_data_access.download(return_query[0]["file_path"]))
         return file_list
 
-    @abstractmethod
+    def upload_products(self, products: list[str]):
+        """
+        Upload data products to the IMAP SDC.
+
+        Attributes
+        ----------
+        products: list[str]
+        A list of file paths to upload to the SDC.
+        """
+        if self.upload_to_sdc:
+            if len(products) == 0:
+                logger.info("No files to upload.")
+            for filename in products:
+                logger.info(f"Uploading file: {filename}")
+                imap_data_access.upload(filename)
```

Comment:
Should we throw an error or warning here if the `upload_to_sdc` attribute is `False?`

Reply:
I think that we could log a warning but I believe that the intent of having the `--upload-to-sdc` option in the CLI is so that users can run processing without uploading products to the SDC by default.